docs/reverse-replication/RunnigReverseReplication.md

```diff
@@ -22,97 +22,202 @@ reverse replication pipeline.
+
+The response looks something like this.
+The Dataflow job ids can be captured to trigger manual shutdown.
+Note the defaults used.
+The gcloud command can be saved to retrigger a specific job if needed.
+
+```
+Setting up reverse replication pipeline...
+metadataInstance not provided, defaulting to target spanner instance id:  <spanner instance id>
+changestream <change stream name> not found
+Creating changestream
+Successfully created changestream <change stream name>
+Created metadata db projects/<project-name>/instances/<instance-name>/databases/<database-name>
+
+GCLOUD CMD FOR READER JOB:
+gcloud dataflow flex-template run smt-reverse-replication-reader-2024-01-05t10-33-56z --project=<project> --region=<region> --template-file-gcs-location=<template location>  --parameters sessionFilePath=<session path>,windowDuration=10s,filtrationMode=forward_migration,skipDirectoryName=skip,instanceId=<spanner instance id>,spannerProjectId=<spanner-project-id>,metadataDatabase=rev_repl_metadata,gcsOutputDirectory=gs://reverse-replication-buffer/data,metadataTableSuffix=,runMode=regular,metadataInstance=<spanner instance>,startTimestamp=,sourceShardsFilePath=<shard file path>,changeStreamName=reverseReplicationStream,databaseId=<spanner database name>,runIdentifier=2024-01-05t10-33-56z --num-workers=5 --worker-machine-type=n2-standard-4 --additional-experiments=use_runner_v2
+
+Launched reader job:  id:"<>" project_id:"<>" name:"<>" current_state_time:{} create_time:{seconds:<> nanos:<>} location:"<region>" start_time:{seconds:<> nanos:<>}
+
+GCLOUD CMD FOR WRITER JOB:
+gcloud dataflow flex-template run smt-reverse-replication-writer-2024-01-05t10-33-56z --project=<project> --region=<region> --template-file-gcs-location=<template location> --parameters sourceShardsFilePath=<shard file path>,metadataTableSuffix=,GCSInputDirectoryPath=gs://reverse-replication-buffer/data,metadataDatabase=rev_repl_metadata,sessionFilePath=<session file path>,sourceDbTimezoneOffset=+00:00,spannerProjectId=<spanner project id>,metadataInstance=<metadata instance>,runMode=regular,runIdentifier=2024-01-05t10-33-56z --num-workers=5 --worker-machine-type=n2-standard-4
+
+Launched writer job:  id:"<>" project_id:"<>" name:"<>" current_state_time:{} create_time:{seconds:<> nanos:<>} location:"<region>" start_time:{seconds:<> nanos:<>}
+
+```
+
+### Custom Jar
+
+In order to specify custom shard identification function, custom jar and class names need to give. The command to do that is below:
+
+```
+go run reverse-replication-runner.go -projectId=<project-id> -dataflowRegion=<region> -instanceId=<spanner-instance> -dbName=<spanner-database> -sourceShardsFilePath=gs://bucket-name/shards.json -sessionFilePath=gs://bucket-name/session.json -readerShardingCustomJarPath=gs://bucket-name/custom.jar -readerShardingCustomClassName=customclassname
```

Comment:
will it better to give example of class name as fully qualified name like com.custom.classname?

Reply:
Done